undertow/src/main/java/org/wildfly/elytron/web/undertow/server/ElytronContextAssociationHandler.java

```diff
@@ -51,7 +52,17 @@ public ElytronContextAssociationHandler(final HttpHandler next, final Supplier<L
      */
     @Override
     public SecurityContext createSecurityContext(HttpServerExchange exchange) {
-        return new SecurityContextImpl(exchange, mechanismSupplier);
+        return new SecurityContextImpl(exchange, mechanismSupplier, createHttpSessionSpi(exchange));
+    }
+
+    /**
+     * Returns a {@link Supplier} that provides a {@link HttpSessionSpi} instance.
+     *
+     * @param exchange the current http exchange
+     * @return a {@link Supplier} that provides a {@link HttpSessionSpi} instance
+     */
+    protected Supplier<HttpSessionSpi> createHttpSessionSpi(HttpServerExchange exchange) {
```

Comment:
This method is all about how session is enabled in WildFly+Undertow and Undertow alone. In case of WildFly, we need to be able to obtain the SessionManager in a different manner.